mobileapps/views.py

```diff
@@ -612,3 +622,238 @@ def _create_notification_message(app_id, payload):
+        """
+        return datetime.datetime.utcnow().replace(tzinfo=utc)
+
+
+class OrganizationThemeView(MobileListCreateAPIView):
+    """
+    **Use Case**
+
+        Get list of themes for organization and create a new theme.
+
+    **Example Requests**
+
+        GET /api/server/mobileapps/organization/{id}/themes
+        POST /api/server/mobileapps/organization/{id}/themes
+
+        **POST Parameters**
+
+        The body of the POST request must include the following parameters.
+
+        * file: Image file to be updated as a logo of the theme
+        * name: Name of the theme (Optional)
+        * active: theme is active or not (Optional)
+
+    **Response Values**
+
+        **GET**
+
+        If the request is successful, the request returns an HTTP 200 "OK" response.
+
+        The HTTP 200 response has a paginated list of objects with the following values.
+
+        * id: ID of the mobile app.
+        * created: Datetime it was created in.
+        * modified: Datetime it was modified in.
+        * logo_image:
+            - has_image
+            - image_url_full
+            - image_url_large
+            - image_url_small
+            - image_url_xsmall
+            - image_url_medium
+        * name: Name of the theme (Optional)
+        * active: theme is active or not (Optional)
+        * organization_id: organization id to which theme is related
+
+        **POST**
+
+        If the request is successful, the request returns an HTTP 204 response.
+
+        * id: ID of the mobile app.
+        * created: Datetime it was created in.
+        * modified: Datetime it was modified in.
+        * logo_image:
+            - has_image
+            - image_url_full
+            - image_url_large
+            - image_url_small
+            - image_url_xsmall
+            - image_url_medium
+        * name: Name of the theme (Optional)
+        * active: theme is active or not (Optional)
+        * organization_id: organization id to which theme is related
+    """
+
+    serializer_class = ThemeSerializer
+
+
+    def get_queryset(self):
+        """
+        Optionally restricts the returned themes to one active only.
+        """
+        return Theme.objects.filter(active=True, organization_id=self.kwargs['organization_id'])
+
+    def post(self, request, organization_id):
+        """
+        POST method inactive the existing active theme and creates and new active one.
+        """
+        data = request.data.copy()
+        data["organization_id"] = organization_id
+
+        Theme.mark_existing_as_inactive(organization_id)
+
+        theme_serializer = ThemeSerializer(data=data)
+        if theme_serializer.is_valid(raise_exception=True):
+            theme = theme_serializer.save()
+
+        organization = Organization.objects.get(pk=organization_id)
+
+        # validate request:
+        # verify that the user's
+        # ensure any file was sent
+        if 'file' not in request.FILES:
```

Comment:
This should be move up to perform validation before performing any transactions.